certbot/plugins/standalone_test.py

```diff
@@ -93,10 +91,10 @@ def test_correct(self):
 
     def test_unrecognized(self):
         assert "foo" not in challenges.Challenge.TYPES
-        self.assertRaises(argparse.ArgumentTypeError, self._call, "foo")
+        self.assertRaises(SystemExit, self._call, "foo")
```

Comment:
`argparse` does a really poor job documenting things, but this is from the docstring at the top of the module:
```
    - ArgumentError -- The exception raised by ArgumentParser objects when
        there are errors with the parser's actions. Errors raised while
        parsing the command-line are caught by ArgumentParser and emitted
        as command-line messages.
```
It causes error messages to look like:
```
$ certbot --standalone-supported-challenges foo
usage: 
  certbot [SUBCOMMAND] [options] [-d DOMAIN] [-d DOMAIN] ...

Certbot can obtain and install HTTPS/TLS/SSL certificates.  By default,
it will attempt to use a webserver both for obtaining and installing the
cert. 
certbot: error: argument --standalone-supported-challenges: Unrecognized challenges: foo
```
rather than something like this when using our own error type:
```
$ certbot --standalone-supported-challenges foo
Unrecognized challenges: foo
```